packages/dds/tree/src/events/events.ts

```diff
@@ -153,22 +154,22 @@ export class EventEmitter<E extends Events<E>> implements IEventEmitter<E> {
      * @returns a function which will deregister the listener when run
      */
     public on<K extends keyof Events<E>>(eventName: K, listener: E[K]): () => void {
-        const listeners = this.listeners[eventName];
+        const listeners = this.listeners.get(eventName);
         if (listeners !== undefined) {
             listeners.add(listener);
         } else {
-            this.listeners[eventName] = new Set([listener]);
+            this.listeners.set(eventName, new Set([listener]));
```

Comment:
our utils have `getOrCreate` which does this.

`getOrCreate(this.listeners, eventName, ()=> new Set()).add(listener)`

that might be cleaner.

Reply:
Lol, I should know that, I think I wrote `getOrCreate()` 😜 . Thanks.